api/v1/harborcluster_types.go

```diff
@@ -344,18 +344,25 @@ type HarborClusterStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 	Conditions []HarborClusterCondition `json:"conditions,omitempty"`
-
-	ComponentsStatus map[Component]*ComponentsStatus `json:"ComponentsStatus,omitempty"`
-}
-
-type ComponentsStatus struct {
-	Replicas      int `json:"replicas,omitempty"`
-	ReadyReplicas int `json:"readyReplicas,omitempty"`
 }
 
 // HarborClusterConditionType is a valid value for HarborClusterConditionType.Type
 type HarborClusterConditionType string
 
+// These are valid conditions of a HarborCluster.
+const (
+	// Ready means the HarborCluster is ready.
+	Ready HarborClusterConditionType = "Ready"
+	// CacheReady means the Cache is ready.
+	CacheReady HarborClusterConditionType = "CacheReady"
+	// DatabaseReady means the Database is ready.
+	DatabaseReady HarborClusterConditionType = "DatabaseReady"
+	// StorageReady means the Storage is ready.
+	StorageReady HarborClusterConditionType = "StorageReady"
+	// ServiceReady means the Service of Harbor is ready.
+	ServiceReady HarborClusterConditionType = "ServiceReady"
```

Comment:
why not use the type of bool directly?

Reply:
It just refer to k8s api-conventions.
`
  Type               FooConditionType   `json:"type" description:"type of Foo condition"`
  Status             ConditionStatus    `json:"status" description:"status of the condition, one of True, False, Unknown"`

  // +optional
  Reason             *string            `json:"reason,omitempty" description:"one-word CamelCase reason for the condition's last transition"`
  // +optional
  Message            *string            `json:"message,omitempty" description:"human-readable message indicating details about last transition"`

  // +optional
  LastHeartbeatTime  *unversioned.Time  `json:"lastHeartbeatTime,omitempty" description:"last time we got an update on a given condition"`
  // +optional
  LastTransitionTime *unversioned.Time  `json:"lastTransitionTime,omitempty" description:"last time the condition transit from one status to another"`

`